lib/honeybadger.ex

```diff
@@ -299,9 +299,19 @@ defmodule Honeybadger do
       |> List.to_string()
     end
 
+    git_revision = fn ->
+      with {_path, 0} <- System.cmd("which", ["git"]),
```

Comment:
Using `with` is great, but we can't rely on `which` for a windows system. While it is doubtful that anybody would be deploying on windows, they may be developing on it.

```suggestion
      with <<_path>> <- System.find_executable("git"),
```

Additionally, what do you think about moving this into a private function? There is a lot happening in the code block.

Reply:
@sorentwo that doesn't work, how about: `with path when is_binary(path) <- System.find_executable("git")` ? cause it returns nil otherwise.

Reply:
Quite right, my mistake! That looks great. 👍